pkg/common/config.go

```diff
@@ -0,0 +1,195 @@
+// Copyright © 2024 Bank-Vaults Maintainers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/spf13/viper"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Config represents the configuration for the webhook
+type Config struct {
+	PspAllowPrivilegeEscalation bool
+	RunAsNonRoot                bool
+	RunAsUser                   int64
+	RunAsGroup                  int64
+	ReadOnlyRootFilesystem      bool
+	RegistrySkipVerify          bool
+	Mutate                      bool
+	MutateProbes                bool
+	Provider                    string
+}
+
+// SecretInitConfig represents the configuration for the secret-init container
+type SecretInitConfig struct {
+	Daemon          bool
+	Delay           time.Duration
+	LogLevel        string
+	JSONLog         string
+	Image           string
+	ImagePullPolicy corev1.PullPolicy
+	LogServer       string
+	CPURequest      resource.Quantity
+	MemoryRequest   resource.Quantity
+	CPULimit        resource.Quantity
+	MemoryLimit     resource.Quantity
+}
+
+func ParseWebhookConfig(obj metav1.Object) Config {
```

Comment:
```suggestion
func LoadWebhookConfig(obj metav1.Object) Config {
```
also replace `Parse` with `Load` for other configs